internal/connector/internal/handlers/processor_secrets.go

```diff
@@ -0,0 +1,36 @@
+package handlers
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/connector/internal/api/dbapi"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/connector/internal/services/vault"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+)
+
+const OwningProcessorResourcePrefix = "/v2alpha1/processor/"
+
+func stripProcessorSecretReferences(resource *dbapi.Processor) *errors.ServiceError {
+	// clear out secrets..
+	resource.ServiceAccount.ClientSecret = ""
+	resource.ServiceAccount.ClientSecretRef = ""
+	return nil
+}
+
+func moveProcessorSecretsToVault(resource *dbapi.Processor, vault vault.VaultService) *errors.ServiceError {
+	// move secrets to a vault.
+	if resource.ServiceAccount.ClientSecret != "" {
+		keyId := api.NewID()
+		if err := vault.SetSecretString(keyId, resource.ServiceAccount.ClientSecret, OwningProcessorResourcePrefix+resource.ID); err != nil {
+			return errors.GeneralError("could not store client secret in the vault: %v", err.Error())
+		}
+		resource.ServiceAccount.ClientSecret = ""
+		resource.ServiceAccount.ClientSecretRef = keyId
+	}
+	return nil
+}
+func getProcessorSecretRefs(resource *dbapi.Processor) (result []string, err error) {
+	if resource.ServiceAccount.ClientSecretRef != "" {
+		result = append(result, resource.ServiceAccount.ClientSecretRef)
+	}
+	return
+}
```

Comment:
This function never returns errors

Reply:
Yes, this is true (and also true of the method [`stripProcessorSecretReferences(..)`](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1686/files#diff-b835efb128f9125006ba8a2c048a61d629283d3f0727169a30574dbc0fdb3bd5R12)).

At this point in time the definition of a `Processor` does not contain secret information (or better said we don't know if it will need to contain secret information) and therefore the two methods here are very basic. However if we need to make the secret handling more complicated it is probable these methods could need to return errors in the future.

I am therefore "future proofing" the code. Less so in this file itself; but in the places where these methods are invoked and we need to handle errors if/should they arise (in the future).

Reply:
In this case I would avoid a `naked return` and would use an explicit one to make it clear that we are always returning `nil`:
```go
return result, nil
```